R/word_splitters.R

```diff
@@ -52,45 +69,67 @@ process_word_recursive <- function(word, current_depth = 1, max_depth = 30) {
     return(word)
   }
 
+  # If we're using the cache and they have this word cached, return that.
+  if (use_lookup) {
+    lookup <- .cache_lookup(cache_dir)
+    if (!is.null(lookup)) {
+      morphemes <- dplyr::filter(lookup, .data$word == .env$word)$morphemes
+      if (length(morphemes)) {
+        return(morphemes[[1]])
+      }
+    }
+  }
+
   # First check for inflections.
-  inf_break <- split_inflections(word)
+  inf_break <- .split_inflections(word)
```

Comment:
since these functions are internal now, may as well make them take the English wikitext rather than the word. We can cut the number of API calls in half if we do this (here and in .split_morphemes).